client/client.go

```diff
@@ -484,7 +486,69 @@ func loadClientTrust(ctx *openssl.Ctx, conf *Config) (*openssl.Ctx, error) {
 	return ctx, nil
 }
 
-func dialOpenSSL(ctx *openssl.Ctx, conf *Config, _ string, addr string) (net.Conn, error) {
+func upgradeTCPConnectionToSSL(
+	ctx *openssl.Ctx,
+	conn net.Conn,
+	addr string,
+	flags openssl.DialFlags,
+) (*openssl.Conn, error) {
+	host, _, err := net.SplitHostPort(addr)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
In the original `openssl.DialSession` there is support for `ctx` being `nil`. Should we add it here also?
```
	if ctx == nil {
		var err error
		ctx, err = NewCtx()
		if err != nil {
			return nil, err
		}
		// TODO: use operating system default certificate chain?
	}
```

Reply:
Why now, The functions are supposed to mirror each other anyhow

Reply:
Resolving convo because function is now in `mendersoftware/openssl`